crates/circuit/src/circuit_instruction.rs

```diff
@@ -440,8 +449,14 @@ impl CircuitInstruction {
                     if blocks_a.len() != blocks_b.len() {
                         return Ok(false);
                     }
+                    // TODO: we should be able to do the semantic-equality comparison from Rust
+                    // space in the future, without going via Python.  See gh-15267.
```

Comment:
Do we have to do this because of the additional data that the Python circuit stores or why? 🤔